extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaSourceModeDialog.js

```diff
@@ -0,0 +1,216 @@
+/*!
+ * VisualEditor user interface WikiaSourceModeDialog class.
+ */
+
+/*global alert, veTrack */
+
+/**
+ * Dialog for editing wikitext in source mode.
+ *
+ * @class
+ * @extends OO.ui.ProcessDialog
+ *
+ * @constructor
+ * @param {Object} [config] Config options
+ */
+ve.ui.WikiaSourceModeDialog = function VeUiWikiaSourceModeDialog( config ) {
+	// Parent constructor
+	ve.ui.WikiaSourceModeDialog.super.call( this, config );
+
+	// Properties
+	this.surface = null;
+};
+
+/* Inheritance */
+
+OO.inheritClass( ve.ui.WikiaSourceModeDialog, ve.ui.FragmentDialog );
+
+/* Static Properties */
+
+ve.ui.WikiaSourceModeDialog.static.name = 'wikiaSourceMode';
+
+ve.ui.WikiaSourceModeDialog.static.title = OO.ui.deferMsg( 'wikia-visualeditor-dialog-wikiasourcemode-title' );
+
+// as in OO.ui.WindowManager.static.sizes
+ve.ui.WikiaSourceModeDialog.static.size = 'large';
+
+ve.ui.WikiaSourceModeDialog.static.icon = 'source';
+
+ve.ui.WikiaSourceModeDialog.static.actions = [
+	{
+		action: 'apply',
+		label: OO.ui.deferMsg( 'visualeditor-dialog-action-apply' ),
+		flags: [ 'progressive', 'primary' ]
+	},
+	{
+		label: OO.ui.deferMsg( 'visualeditor-dialog-action-cancel' ),
+		flags: 'safe'
+	}
+];
+
+/* Methods */
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaSourceModeDialog.prototype.getBodyHeight = function () {
+	return 600;
+};
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaSourceModeDialog.prototype.initialize = function () {
+	// Parent method
+	ve.ui.WikiaSourceModeDialog.super.prototype.initialize.call( this );
+
+	// Properties
+	this.openCount = 0; // tracking
+	this.timings = {}; // tracking
+	this.sourceModeTextarea = new OO.ui.TextInputWidget({
+		$: this.$,
+		multiline: true
+	});
+
+	this.initLinkSuggest();
+	this.$body.append( this.sourceModeTextarea.$element );
+	this.$content.addClass( 've-ui-wikiaSourceModeDialog-content' );
+};
+
+/**
+ * Initialize link suggest (if available)
+ *
+ * @method
+ */
+ve.ui.WikiaSourceModeDialog.prototype.initLinkSuggest = function () {
+	if ( !!mw.loader.getState( 'ext.wikia.LinkSuggest' ) ) {
+		// jquery.ui.autocomplete.js
+		this.sourceModeTextarea.$input.css( {
+			'z-index': 9999999,
+			position: 'relative'
+		} );
+		mw.loader.using(
+			'ext.wikia.LinkSuggest',
+			function () {
+				this.sourceModeTextarea.$input.linksuggest();
```

Comment:
Was done like this because `this.sourceModeTextarea.$input.linksuggest` does not exist until after `ext.wikia.LinkSuggest` has loaded so it cannot be bound to.